### PR TITLE
Add `map` type support inside `union`

### PR DESCRIPTION
**Pull request by isenilov**
This PR adds support of `map` type inside union.
Schema excerpt that caused the error at https://github.com/SRserves85/avro-to-python/blob/bacb3e6eb8bb09bb0a4702f6a5f48a3ef89e11f2/avro_to_python/utils/avro/types/union.py#L101:

```json
...,
{
  "name": "some_name",
  "type": [
    "null",
    {
      "type": "map",
      "values": "float"
    }
  ],
  "default": null
},
...
```

@SRserves85 @irux @chasdevs Please let me know if this makes sense so I could add the tests for the case.

### CODE REVIEW

**Review by chasdevs — APPROVED** [review]
Been a while for me so I forget if there's another piece you need to implement, but the changes look fine to me. Tests _should_ identify if anything is missing.

**Comment by isenilov** [comment]
Thank you @chasdevs ! I guess you also need to approve the workflow if it is ok to merge.

**Comment by chasdevs** [comment]
No problem; would you mind adding a quick test before the merge? If it's an emergency I can approve but want to keep that coverage high.

**Comment by chasdevs** [comment]
Hmm let me try that out rq. Thanks for adding the test data!

**Comment by isenilov** [comment]
Seems like it works!

**Pull request merged**
